drivers/sensor/st/lsm6dsv16x/lsm6dsv16x_trigger.c

```diff
@@ -177,7 +177,11 @@ static void lsm6dsv16x_handle_interrupt(const struct device *dev)
 	int ret;
 
 	while (1) {
-		if (IS_ENABLED(CONFIG_LSM6DSV16X_STREAM)) {
+		/* When using I3C IBI interrupt the status register is already automatically
+		 * read (clearing the interrupt condition), so we can skip the extra bus
+		 * transaction for FIFO stream case.
+		 */
+		if (ON_I3C_BUS(cfg) && I3C_INT_PIN(cfg) && IS_ENABLED(CONFIG_LSM6DSV16X_STREAM)) {
```

Comment:
This will not work if it is using IBI due to the wrong check on `I3C_INT_PIN`. Should be
```c
if (ON_I3C_BUS(cfg) && !I3C_INT_PIN(cfg) && IS_ENABLED(CONFIG_LSM6DSV16X_STREAM)) {
```

Reply:
ok, my bad. :(